src/core/managers/chat_manager.cpp

```diff
@@ -68,16 +68,17 @@ void DetourHostSay(CBaseEntity* pController, CCommand& args, bool teamonly, int
         }
     }
 
-    bool bSilent = globals::coreConfig->SilentChatTrigger.find(*args[1]) != std::string::npos;
-    bool bCommand = globals::coreConfig->PublicChatTrigger.find(*args[1]) != std::string::npos || bSilent;
+    std::string* prefix;
+    bool bSilent = globals::coreConfig->IsSilentChatTrigger(args[1], prefix);
+    bool bCommand = bSilent || globals::coreConfig->IsPublicChatTrigger(args[1], prefix);
 
     if (!bSilent) {
         m_pHostSay(pController, args, teamonly, unk1, unk2);
     }
 
     if (bCommand)
     {
-        char *pszMessage = (char *)(args.ArgS() + 2);
+        char *pszMessage = (char *)(args.ArgS() + prefix->length() + 1);

```

Comment:
I think this needs further tests as I had no time but it was working fine for me, but had crashes in the early state of the implementation of this part

Reply:
okay it seems pretty fine